tests/testImageMapReduce.py

```diff
@@ -0,0 +1,243 @@
+        @param[in] fullBBox the bounding box of the original exposure
+        @return a float, the mean of `subExp.getMaskedImage().getArray()`.
+        """
+        return subExp.getMaskedImage().getImage().getArray().mean()
+
+
+class TestImageMapReduceConfig2(ImageMapReduceConfig):
+    """!
+    \anchor TestImageMapReduceConfig_
+
+    \brief Configuration parameters for the TestImageMapReduceTask2
+    """
+    mapperSubtask = pexConfig.ConfigurableField(
+        doc="Mapper subtask to run on each subimage",
+        target=TestImageMapperSubtask2,
+    )
+
+
+class ImageMapReduceTest(lsst.utils.tests.TestCase):
+    """!A test case for the image gridded processing task
+    """
+
+    def setUp(self):
+        self.exposure = afwImage.ExposureF(128, 128)
+        self.exposure.setPsf(measAlg.DoubleGaussianPsf(11, 11, 2.0, 3.7))
+        mi = self.exposure.getMaskedImage()
+        mi[:, :] = 0.
+
+    def tearDown(self):
+        del self.exposure
+
+    def testExampleTaskNoOverlaps(self):
+        """Test sample grid task that adds 5.0 to input image and uses
+        default 'copy' `reduceOperation`.
+        """
+        config = TestImageMapReduceConfig()
+        task = ImageMapReduceTask(config)
+        config.mapperSubtask.addAmount = 5.
+        newExp = task.run(self.exposure)
+        newMI = newExp.getMaskedImage()
+        newArr = newMI.getImage().getArray()
+        mi = self.exposure.getMaskedImage()
+
+        self.assertClose(mi.getImage().getArray().mean(), newArr.mean() - 5.)
+        self.assertClose(newArr.mean(), 5.)
+        self.assertClose(newArr.min(), 5.)
+        self.assertClose(newArr.max(), 5.)
+
+    def testExampleTaskWithOverlaps(self):
+        """Test sample grid task that adds 5.0 to input image and uses
+        'average' `reduceOperation`.
+        """
+        config = TestImageMapReduceConfig()
+        config.gridStepX = config.gridStepY = 8
+        config.reducerSubtask.reduceOperation = 'average'
+        task = ImageMapReduceTask(config)
+        config.mapperSubtask.addAmount = 5.
+        newExp = task.run(self.exposure)
+        newMI = newExp.getMaskedImage()
+        newArr = newMI.getImage().getArray()
+        mi = self.exposure.getMaskedImage()
+
+        self.assertClose(mi.getImage().getArray().mean(), newArr.mean() - 5.)
+        self.assertClose(newArr.mean(), 5.)
+        self.assertClose(newArr.min(), 5.)
+        self.assertClose(newArr.max(), 5.)
+
+    def testExampleTaskMean(self):
+        """Test sample grid task that returns the mean of the subimages and uses
+        'none' `reduceOperation`.
+        """
+        config = TestImageMapReduceConfig2()
+        config.gridStepX = config.gridStepY = 8
+        config.reducerSubtask.reduceOperation = 'none'
+        task = ImageMapReduceTask(config)
+        subMeans = task.run(self.exposure)
+
+        self.assertEqual(len(subMeans), len(task.boxes0))
+        self.assertClose(subMeans[0], 0.)
```

Comment:
I don't know what `subMeans` should look like, but you should test all values. If it really should be all zeros, use `self.assertFloatsEqual(subMeans, 0)`. Although, I wonder if you shouldn't have some values in it, so that the mean is something more interesting than zero?